src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -79,18 +81,27 @@
      *
      * @param modelId   Id of model to create
      * @param model Model to be indexed
-     * @param listener  handles acknowledged response
+     * @param listener  handles index response
      */
-    void put(String modelId, Model model, ActionListener<AcknowledgedResponse> listener) throws IOException;
+    void put(String modelId, Model model, ActionListener<IndexResponse> listener) throws IOException;
 
     /**
      * Put a model into the system index. Non-blocking. When no id is passed in, OpenSearch will generate the id
      * automatically. The id can be retrieved in the IndexResponse.
      *
      * @param model Model to be indexed
-     * @param listener  handles acknowledged response
+     * @param listener  handles index response
      */
-    void put(Model model, ActionListener<AcknowledgedResponse> listener) throws IOException;
+    void put(Model model, ActionListener<IndexResponse> listener) throws IOException;
+
+    /**
+     * Update model of model id with new model.
+     *
+     * @param modelId model id to update
+     * @param model new model
+     * @param listener handles index response
+     */
+    void update(String modelId, Model model, ActionListener<IndexResponse> listener) throws IOException;
```

Comment:
I thought we do not want to update ```model``` once created? Can some one do it using this function?

Reply:
This is required because initially we set state as training, then we train, then we update state to created and add the model. This will not be exposed to user.